src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -429,16 +428,33 @@ private static bool ShouldUseInlineArray(BoundCollectionExpressionBase node, CSh
             SyntaxNode syntax,
             TypeWithAnnotations elementType,
             ImmutableArray<BoundNode> elements,
-            bool asReadOnlySpan,
-            ArrayBuilder<LocalSymbol> locals)
+            bool asReadOnlySpan)
         {
             Debug.Assert(elements.Length > 0);
             Debug.Assert(elements.All(e => e is BoundExpression));
             Debug.Assert(_factory.ModuleBuilderOpt is { });
             Debug.Assert(_diagnostics.DiagnosticBag is { });
             Debug.Assert(_compilation.Assembly.RuntimeSupportsInlineArrayTypes);
+            Debug.Assert(_additionalLocals is { });
 
             int arrayLength = elements.Length;
+            if (arrayLength == 1
+                && _factory.WellKnownMember(asReadOnlySpan
+                    ? WellKnownMember.System_ReadOnlySpan_T__ctor_ref_readonly_T
+                    : WellKnownMember.System_Span_T__ctor_ref_T, isOptional: true) is MethodSymbol spanRefConstructor)
+            {
+                // Special case: no need to create an InlineArray1 type. Just use a temp of the element type.
+                var spanType = _factory
+                    .WellKnownType(asReadOnlySpan ? WellKnownType.System_ReadOnlySpan_T : WellKnownType.System_Span_T)
+                    .Construct([elementType]);
+                var constructor = spanRefConstructor.AsMember(spanType);
+                var element = VisitExpression((BoundExpression)elements[0]);
+                var temp = _factory.StoreToTemp(element, out var assignment);
+                _additionalLocals.Add(temp.LocalSymbol);
```

Comment:
This feels likely to me to have some potentially bad interactions with the work @jjonescz is doing for allowing ref locals in async methods; seems possible that we'll end up trying to lift these locals into closure state. Jan and Rikki, is there any coordination between this and the feature branch that we need to do to avoid issues?

Reply:
> This feels likely to me to have some potentially bad interactions with the work @jjonescz is doing for allowing ref locals in async methods; seems possible that we'll end up trying to lift these locals into closure state.

The local that is added to the method here is not a ref local and is not a ref struct. Therefore, I do not think there will be any issues specifically with lifting that local.

Reply:
Thanks for the clarification Aleksey.